projects/api/src/brand/brand.service.ts

```diff
@@ -0,0 +1,76 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { Brand } from '@prisma/client';
+
+import PrismaService from '@/prisma/prisma.service';
+
+import CreateBrandDto from './dto/create-brand.dto';
+import UpdateBrandDto from './dto/update-brand.dto';
+
+@Injectable()
+class BrandService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(createBrandDto: CreateBrandDto): Promise<Brand> {
+    try {
+      return await this.prisma.brand.create({
+        data: createBrandDto,
+      });
+    } catch (error) {
+      if (error?.meta?.target?.includes('name')) {
+        throw new BadRequestException('Brand name already exists');
```

Comment:
If you dont put in your model that this field name is unique, you don't need to validate this possible error, but we need that the name be unique, so update your model please